ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,178 @@
+
+    date = forms.DateField(
+        label='Date of reservation (mm/dd/yy)',
+        widget=forms.DateInput(attrs={'placeholder': '02/28/18'}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (xx:xx)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (xx:xx)',
+    )
+
+    disclaimer_agreement = forms.BooleanField(
+        label='I agree with the above statement.',
+        error_messages={
+            'required': 'You must agree to our policies.',
+        },
+    )
+
+    def clean_verify_contact_email(self):
+        email = self.cleaned_data.get('contact_email')
+        verify_contact_email = self.cleaned_data.get('verify_contact_email')
+
+        if email and verify_contact_email:
+            if email != verify_contact_email:
+                raise forms.ValidationError("Your emails don't match.")
+        return verify_contact_email
+
+
+@calnet_required
+def request_reservation(request):
+    calnet_uid = request.session['calnet_uid']
+    status = 'new_request'
+
+    # ensure we can even find them in university LDAP
+    # (alumni etc. might not be readable in LDAP but can still auth via CalNet)
+    if not user_attrs_ucb(calnet_uid):
+        return render(
+            request,
+            'lab_reservations/lab_reservations/cant-find-in-ldap.html',
+            {
+                'calnet_uid': calnet_uid,
+                'title': 'Unable to read account information',
+            },
+        )
+
+    if request.method == 'POST':
+        form = RequestForm(request.POST)
+        if form.is_valid():
+            req = NewReservationRequest(
+                real_name=form.cleaned_data['real_name'],
+                contact_email=form.cleaned_data['contact_email'],
+                student_group=form.cleaned_data['student_group'],
+                reason=form.cleaned_data['reason'],
+                date=form.cleaned_data['date'],
+                starttime=form.cleaned_data['starttime'],
+                endtime=form.cleaned_data['endtime'],
+            )
+
+            send_request_to_officers(req)
+            send_request_confirmation(req)
+
+            return render(
+                request,
+                'lab_reservations/pending.html',
+            )
+
+    else:
+        form = RequestForm()
+
+    return render(
+        request,
+        'lab_reservations/index.html',
+        {
+            'form': form,
+            'status': status,
+            'title': 'Request the OCF Lab Space',
+        },
+    )
+
+
+def reservation_requested(request):
+    return render(request, 'lab_reservations/pending.html', {'title': 'Reservation request successful'})
+
+
+def send_request_to_officers(request):
+    body = JINJA_MAIL_ENV.get_template(
+        'lab_reservations/mail_templates/officer_notification.jinja',
+    ).render(request=request)
+    send_mail('bod@ocf.berkeley.edu', 'New Lab Reservation Request', body, sender=request.contact_email)
```

Comment:
Maybe add some additional info to the subject? If we get multiple submissions at a time, it would help if the subject was `Lab Reservation Request: *Group Name*`, for example, rather than just many submissions with the same subject.